lib/android_build/maesdk/src/main/java/com/microsoft/applications/events/HttpClient.java

```diff
@@ -210,14 +210,23 @@ public HttpClient(Context context) {
     if (hasConnectivityManager()) {
       if (context.checkSelfPermission(permission.ACCESS_NETWORK_STATE)
           == PackageManager.PERMISSION_GRANTED) {
-        m_connectivityManager =
-            (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
-        if (m_connectivityManager != null) {
-          boolean is_metered = m_connectivityManager.isActiveNetworkMetered();
-          m_callback = new ConnectivityCallback(this, is_metered);
-          onCostChange(is_metered); // set initial value in C++ side
-          m_connectivityManager.registerDefaultNetworkCallback(m_callback);
-        }
+            try {
+              m_connectivityManager =
+                  (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
+              if (m_connectivityManager != null) {
+                boolean is_metered = m_connectivityManager.isActiveNetworkMetered();
+                m_callback = new ConnectivityCallback(this, is_metered);
+                onCostChange(is_metered); // set initial value in C++ side
+                m_connectivityManager.registerDefaultNetworkCallback(m_callback);
+              }
+            }
+            catch (Exception e) {
```

Comment:
Should we explicitly catch the below known exceptions, rather than generically catching and throwing away everything?

Reply:
I think catching generic one is very good: that secures us from anything bad happening. But maybe we need to log that somewhere, in logcat.

Reply:
I split out the different exceptions in case we want to handle them separately in the future + added logs for all of them in the latest iteration.